Implemented cleanup portion in test-pvc-deletion test case and deploying pvc in different namespace

gprasath opened the pull request:
Signed-off-by: gprasath <giridhara.prasad@cloudbyte.com>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:

  - Obtain the number of nodes and deploy one replica more than the node count.
  - Create PVC in non default namespace.
  - Check if there is a replica in pending state.
  - Deleting pvc deployment.
  - Check if the pv and svc are deleted successfully.
  - Clean the remaining resources and test artifacts.

**Special notes for your reviewer**:

  - Creating a namespace pvc-deletion.
  - Deploying app in that namespace.
  - After successful verification, deleting the pvc.
  - Check if the relevant resources are deleted.
  - Cleaning up the resources.

gprasath commented:
Sample test-run output:

```
giri@vagrant-host:~/openebs/e2e/ansible$ ansible-playbook playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml -vv
ansible-playbook 2.4.3.0
  config file = /home/giri/openebs/e2e/ansible/ansible.cfg
  configured module search path = [u'/home/giri/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
Using /home/giri/openebs/e2e/ansible/ansible.cfg as config file
[DEPRECATION WARNING]: The use of 'include' for tasks has been deprecated. Use 'import_tasks' for static inclusions or 'include_tasks' for dynamic inclusions. This
feature will be removed in a future release. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: include is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this
rationale.. This feature will be removed in a future release. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
statically imported: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/cleanup.yml
statically imported: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/pre-requisites.yml

PLAYBOOK: k8s_delete_pvc.yml *******************************************************************************************************************************************
1 plays in playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml

PLAY [localhost] *******************************************************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************************************************************
ok: [localhost]
META: ran handlers

TASK [Get $HOME of K8s master for kubernetes user] *********************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/pre-requisites.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; echo $HOME", "delta": "0:00:00.010137", "end": "2018-05-29 13:10:15.876639", "rc": 0, "start": "2018-05-29 13:10:15.866502", "stderr": "", "stderr_lines": [], "stdout": "/home/ubuntu", "stdout_lines": ["/home/ubuntu"]}

TASK [Start the log aggregator to capture test pod logs] ***************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/pre-requisites.yml:10
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; nohup stern \"maya*|openebs*|pvc*|\" --since 1m > \"/home/ubuntu/setup/logs/delete_pvc.log\" &", "delta": "0:00:00.229546", "end": "2018-05-29 13:10:16.943600", "rc": 0, "start": "2018-05-29 13:10:16.714054", "stderr": "mesg: ttyname failed: Inappropriate ioctl for device", "stderr_lines": ["mesg: ttyname failed: Inappropriate ioctl for device"], "stdout": "", "stdout_lines": []}

TASK [2) Get the number of nodes in the cluster] ***********************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:33
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get nodes | grep '<none>' | wc -l", "delta": "0:00:01.503697", "end": "2018-05-29 13:10:19.120907", "rc": 0, "start": "2018-05-29 13:10:17.617210", "stderr": "", "stderr_lines": [], "stdout": "3", "stdout_lines": ["3"]}

TASK [2a) Fetch the node count from stdout] ****************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:40
ok: [localhost] => {"ansible_facts": {"node_count": " 3"}, "changed": false}

TASK [3) Copy the storage classes yaml to k8s master] ******************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:45
changed: [localhost -> None] => {"changed": true, "checksum": "befb15bd893b94f43ce51486b6b84e476c476651", "dest": "/home/ubuntu/storage-class.yaml", "gid": 1000, "group": "ubuntu", "md5sum": "f7215335557c5d80140c713e05b81a84", "mode": "0664", "owner": "ubuntu", "size": 272, "src": "/home/ubuntu/.ansible/tmp/ansible-tmp-1527599419.46-7161758241147/source", "state": "file", "uid": 1000}

TASK [3a) Replace the replica count in storage class file] *************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:52
ok: [localhost -> None] => {"changed": false, "msg": ""}

TASK [4) Check whether maya-apiserver pod is deployed] *****************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:59
included: /home/giri/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check openebs-api pod status] ************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n openebs | grep openebs-api", "delta": "0:00:01.615309", "end": "2018-05-29 13:10:26.910311", "rc": 0, "start": "2018-05-29 13:10:25.295002", "stderr": "", "stderr_lines": [], "stdout": "openebs-apiserver-68cbb698d-dh69g      1/1       Running   0          5h", "stdout_lines": ["openebs-apiserver-68cbb698d-dh69g      1/1       Running   0          5h"]}

TASK [5) Create test specific namespace] *******************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:65
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl create ns pvc-deletion", "delta": "0:00:00.768326", "end": "2018-05-29 13:10:28.782785", "rc": 0, "start": "2018-05-29 13:10:28.014459", "stderr": "", "stderr_lines": [], "stdout": "namespace \"pvc-deletion\" created", "stdout_lines": ["namespace \"pvc-deletion\" created"]}

TASK [6) Create the storage class in K8s master] ***********************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:72
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f \"storage-class.yaml\"", "delta": "0:00:00.830296", "end": "2018-05-29 13:10:30.975151", "rc": 0, "start": "2018-05-29 13:10:30.144855", "stderr": "", "stderr_lines": [], "stdout": "storageclass \"openebs-test\" configured", "stdout_lines": ["storageclass \"openebs-test\" configured"]}

TASK [6a) Copy pvc yaml to K8s master] *********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:79
changed: [localhost -> None] => {"changed": true, "checksum": "4964d9e88fb6a00362530a948747ab35ab635122", "dest": "/home/ubuntu/create_pvc.yaml", "gid": 1000, "group": "ubuntu", "md5sum": "8230388d76de6c486401796e602126ab", "mode": "0664", "owner": "ubuntu", "size": 195, "src": "/home/ubuntu/.ansible/tmp/ansible-tmp-1527599431.27-264727823069202/source", "state": "file", "uid": 1000}

TASK [6b) Create PVC in k8s cluster] ***********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:85
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl apply -f \"create_pvc.yaml\" -n \"pvc-deletion\"", "delta": "0:00:00.969786", "end": "2018-05-29 13:10:35.936239", "rc": 0, "start": "2018-05-29 13:10:34.966453", "stderr": "", "stderr_lines": [], "stdout": "persistentvolumeclaim \"openebsvolume\" created", "stdout_lines": ["persistentvolumeclaim \"openebsvolume\" created"]}

TASK [Wait for few seconds] ********************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:91
ok: [localhost -> None] => {"changed": false, "elapsed": 30, "path": null, "port": null, "search_regex": null, "state": "started"}

TASK [7) Finding PV name] **********************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:96
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get pv | grep \"openebs-test\" | awk {'print $1'}", "delta": "0:00:00.612851", "end": "2018-05-29 13:11:09.110654", "rc": 0, "start": "2018-05-29 13:11:08.497803", "stderr": "", "stderr_lines": [], "stdout": "pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db", "stdout_lines": ["pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db"]}

TASK [7a) Confirm whether the volume pods are created] *****************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:104
changed: [localhost -> None] => (item=ctrl) => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n pvc-deletion | grep \"pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db\" | grep ctrl | wc -l", "delta": "0:00:01.142527", "end": "2018-05-29 13:11:11.528021", "item": "ctrl", "rc": 0, "start": "2018-05-29 13:11:10.385494", "stderr": "", "stderr_lines": [], "stdout": "1", "stdout_lines": ["1"]}
changed: [localhost -> None] => (item=rep) => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n pvc-deletion | grep \"pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db\" | grep rep | wc -l", "delta": "0:00:01.540586", "end": "2018-05-29 13:11:13.812974", "item": "rep", "rc": 0, "start": "2018-05-29 13:11:12.272388", "stderr": "", "stderr_lines": [], "stdout": "4", "stdout_lines": ["4"]}

TASK [6) Check if one replica is unscheduled] **************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:119
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n pvc-deletion| grep \"pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db\" | grep -i \"pending\" |wc -l", "delta": "0:00:00.815880", "end": "2018-05-29 13:11:15.611880", "rc": 0, "start": "2018-05-29 13:11:14.796000", "stderr": "", "stderr_lines": [], "stdout": "1", "stdout_lines": ["1"]}

TASK [6b) Check if the svc is created] *********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:131
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get svc -n pvc-deletion | grep \"pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db\" |wc -l", "delta": "0:00:00.459097", "end": "2018-05-29 13:11:16.667433", "rc": 0, "start": "2018-05-29 13:11:16.208336", "stderr": "", "stderr_lines": [], "stdout": "1", "stdout_lines": ["1"]}

TASK [6c) Getting service name] ****************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:142
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get svc -n pvc-deletion| grep \"pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db\" | awk {'print $1'}", "delta": "0:00:01.058989", "end": "2018-05-29 13:11:18.424004", "rc": 0, "start": "2018-05-29 13:11:17.365015", "stderr": "", "stderr_lines": [], "stdout": "pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db-ctrl-svc", "stdout_lines": ["pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db-ctrl-svc"]}

TASK [7) Delete PVC] ***************************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:149
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl delete -f \"create_pvc.yaml\" -n pvc-deletion", "delta": "0:00:00.526470", "end": "2018-05-29 13:11:20.006439", "rc": 0, "start": "2018-05-29 13:11:19.479969", "stderr": "", "stderr_lines": [], "stdout": "persistentvolumeclaim \"openebsvolume\" deleted", "stdout_lines": ["persistentvolumeclaim \"openebsvolume\" deleted"]}

TASK [8) check if the pods are deleted] ********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:155
FAILED - RETRYING: 8) check if the pods are deleted (10 retries left).
changed: [localhost -> None] => (item=ctrl) => {"attempts": 2, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n pvc-deletion | grep \"pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db\" | grep ctrl | wc -l", "delta": "0:00:00.484625", "end": "2018-05-29 13:13:25.427713", "item": "ctrl", "rc": 0, "start": "2018-05-29 13:13:24.943088", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "0", "stdout_lines": ["0"]}
changed: [localhost -> None] => (item=rep) => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n pvc-deletion | grep \"pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db\" | grep rep | wc -l", "delta": "0:00:00.545288", "end": "2018-05-29 13:13:27.003247", "item": "rep", "rc": 0, "start": "2018-05-29 13:13:26.457959", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "0", "stdout_lines": ["0"]}

TASK [8a) Check if the pv is deleted] **********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:168
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: '{{ pv_name.stdout }}' not in pv_out.stdout

changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pv", "delta": "0:00:00.758033", "end": "2018-05-29 13:13:28.987563", "rc": 0, "start": "2018-05-29 13:13:28.229530", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "", "stdout_lines": []}

TASK [8b) Check if the svc is deleted] *********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:178
FAILED - RETRYING: 8b) Check if the svc is deleted (5 retries left).
FAILED - RETRYING: 8b) Check if the svc is deleted (4 retries left).
FAILED - RETRYING: 8b) Check if the svc is deleted (3 retries left).
FAILED - RETRYING: 8b) Check if the svc is deleted (2 retries left).
FAILED - RETRYING: 8b) Check if the svc is deleted (1 retries left).
fatal: [localhost -> None]: FAILED! => {"attempts": 5, "changed": true, "cmd": "source ~/.profile; kubectl get svc -n pvc-deletion | grep \"pvc-acdd61d0-6341-11e8-94dd-02b983f0a4db-ctrl-svc\" | wc -l", "delta": "0:00:00.789542", "end": "2018-05-29 13:15:20.293241", "rc": 0, "start": "2018-05-29 13:15:19.503699", "stderr": "", "stderr_lines": [], "stdout": "1", "stdout_lines": ["1"]}

TASK [Setting failed flag] *********************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:193
ok: [localhost] => {"ansible_facts": {"flag": "TEST FAILED"}, "changed": false}

TASK [Delete the namespace] ********************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/cleanup.yml:2
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl delete ns pvc-deletion", "delta": "0:00:00.787100", "end": "2018-05-29 13:15:22.219213", "rc": 0, "start": "2018-05-29 13:15:21.432113", "stderr": "", "stderr_lines": [], "stdout": "namespace \"pvc-deletion\" deleted", "stdout_lines": ["namespace \"pvc-deletion\" deleted"]}

TASK [Remove the test artifacts] ***************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/cleanup.yml:12
changed: [localhost -> None] => (item=storage-class.yaml) => {"changed": true, "item": "storage-class.yaml", "path": "/home/ubuntu/storage-class.yaml", "state": "absent"}
changed: [localhost -> None] => (item=create_pvc.yaml) => {"changed": true, "item": "create_pvc.yaml", "path": "/home/ubuntu/create_pvc.yaml", "state": "absent"}

TASK [Setting cleanup flag to passed] **********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:204
ok: [localhost] => {"ansible_facts": {"cflag": "CLEANUP PASSED"}, "changed": false}

TASK [5) Terminate the log aggregator] *********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:216
fatal: [localhost -> None]: FAILED! => {"changed": true, "cmd": "source ~/.profile; killall stern", "delta": "0:00:00.026671", "end": "2018-05-29 13:15:24.766654", "msg": "non-zero return code", "rc": 1, "start": "2018-05-29 13:15:24.739983", "stderr": "stern: no process found", "stderr_lines": ["stern: no process found"], "stdout": "", "stdout_lines": []}

PLAY RECAP *************************************************************************************************************************************************************
localhost                  : ok=26   changed=19   unreachable=0    failed=2
```

dargasudarshan commented on e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:
```diff
@@ -21,176 +24,205 @@
-         args:
-           executable: /bin/bash
-         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+        - name: 6) Create the storage class in K8s master
```
@gprasath Can you include the task to deploy applications?

gprasath replied:
Addressed.

dargasudarshan commented on e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:
```diff
@@ -21,176 +24,205 @@
        
+          always:
+        
+            - name: 5) Terminate the log aggregator
```
For Stern Start/kill also we have tasks @gprasath Can we use them?

yudaykiran commented on e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:
```diff
@@ -21,176 +24,202 @@
+# Attempting to create more replicas than the number of nodes
+        - name: 3a) Replace the replica count in storage class file
+          replace:
+            path: storage-class.yaml
```
is $HOME/storage-class.yaml required here?

gprasath commented:
Test run output:

```
giri@vagrant-host:~/openebs/e2e/ansible$ ansible-playbook playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml -vv
ansible-playbook 2.4.3.0
  config file = /home/giri/openebs/e2e/ansible/ansible.cfg
  configured module search path = [u'/home/giri/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
Using /home/giri/openebs/e2e/ansible/ansible.cfg as config file
[DEPRECATION WARNING]: The use of 'include' for tasks has been deprecated. Use 'import_tasks' for static inclusions or 'include_tasks' for dynamic inclusions. This
feature will be removed in a future release. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: include is kept for backwards compatibility but usage is discouraged. The module documentation details page may explain more about this
rationale.. This feature will be removed in a future release. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
statically imported: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/cleanup.yml
statically imported: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/pre-requisites.yml

PLAYBOOK: k8s_delete_pvc.yml *******************************************************************************************************************************************
1 plays in playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml

PLAY [localhost] *******************************************************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************************************************************
ok: [localhost]
META: ran handlers

TASK [Get $HOME of K8s master for kubernetes user] *********************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/pre-requisites.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; echo $HOME", "delta": "0:00:00.012465", "end": "2018-06-12 11:43:53.242473", "rc": 0, "start": "2018-06-12 11:43:53.230008", "stderr": "", "stderr_lines": [], "stdout": "/home/ubuntu", "stdout_lines": ["/home/ubuntu"]}

TASK [include_tasks] ***************************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/pre-requisites.yml:10
included: /home/giri/openebs/e2e/ansible/playbooks/utils/stern_task.yml for localhost

TASK [Start the log aggregator to capture test pod logs] ***************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/utils/stern_task.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; nohup stern --all-namespaces \"maya*|openebs*|pvc*|\" --kubeconfig /home/ubuntu/.kube/config --since 1m > /home/ubuntu/setup/logs/delete_pvc.log &", "delta": "0:00:01.019957", "end": "2018-06-12 11:43:55.237952", "rc": 0, "start": "2018-06-12 11:43:54.217995", "stderr": "mesg: ttyname failed: Inappropriate ioctl for device", "stderr_lines": ["mesg: ttyname failed: Inappropriate ioctl for device"], "stdout": "", "stdout_lines": []}

TASK [Terminate the log aggregator] ************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/utils/stern_task.yml:14
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [2) Get the number of nodes in the cluster] ***********************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:33
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get nodes | grep '<none>' | wc -l", "delta": "0:00:01.368882", "end": "2018-06-12 11:43:57.506477", "rc": 0, "start": "2018-06-12 11:43:56.137595", "stderr": "", "stderr_lines": [], "stdout": "3", "stdout_lines": ["3"]}

TASK [2a) Fetch the node count from stdout] ****************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:40
ok: [localhost] => {"ansible_facts": {"node_count": " 3"}, "changed": false}

TASK [3) Copy the storage classes yaml to k8s master] ******************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:45
changed: [localhost -> None] => {"changed": true, "checksum": "f2f5eace352e011490addf347d48f2101e115b5a", "dest": "/home/ubuntu/storage-class.yaml", "gid": 1000, "group": "ubuntu", "md5sum": "d900a45f2f6f7b6a2727012a0fb201c3", "mode": "0664", "owner": "ubuntu", "size": 272, "src": "/home/ubuntu/.ansible/tmp/ansible-tmp-1528803838.86-151412545169910/source", "state": "file", "uid": 1000}

TASK [3a) Replace the replica count in storage class file] *************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:52
changed: [localhost -> None] => {"changed": true, "msg": "1 replacements made"}

TASK [4) Check whether maya-apiserver pod is deployed] *****************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:59
included: /home/giri/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check apiserver pod status] **************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n openebs | grep apiserver", "delta": "0:00:01.115113", "end": "2018-06-12 11:44:03.742646", "rc": 0, "start": "2018-06-12 11:44:02.627533", "stderr": "", "stderr_lines": [], "stdout": "maya-apiserver-84bfd9cc5b-qxg64        1/1       Running   2          4h", "stdout_lines": ["maya-apiserver-84bfd9cc5b-qxg64        1/1       Running   2          4h"]}

TASK [5) Create test specific namespace] *******************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:65
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl create ns pvc-deletion", "delta": "0:00:00.493356", "end": "2018-06-12 11:44:05.550152", "rc": 0, "start": "2018-06-12 11:44:05.056796", "stderr": "", "stderr_lines": [], "stdout": "namespace \"pvc-deletion\" created", "stdout_lines": ["namespace \"pvc-deletion\" created"]}

TASK [6) Create the storage class in K8s master] ***********************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:72
included: /home/giri/openebs/e2e/ansible/playbooks/utils/deploy_task.yml for localhost

TASK [Deploy application files] ****************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/utils/deploy_task.yml:2
changed: [localhost -> None] => (item=/home/ubuntu/storage-class.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f /home/ubuntu/storage-class.yaml -n default", "delta": "0:00:00.695897", "end": "2018-06-12 11:44:06.992178", "item": "/home/ubuntu/storage-class.yaml", "rc": 0, "start": "2018-06-12 11:44:06.296281", "stderr": "", "stderr_lines": [], "stdout": "storageclass.storage.k8s.io \"openebs-test\" created", "stdout_lines": ["storageclass.storage.k8s.io \"openebs-test\" created"]}

TASK [6a) Copy pvc yaml to K8s master] *********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:79
changed: [localhost -> None] => {"changed": true, "checksum": "4964d9e88fb6a00362530a948747ab35ab635122", "dest": "/home/ubuntu/create_pvc.yaml", "gid": 1000, "group": "ubuntu", "md5sum": "8230388d76de6c486401796e602126ab", "mode": "0664", "owner": "ubuntu", "size": 195, "src": "/home/ubuntu/.ansible/tmp/ansible-tmp-1528803848.12-204039839864361/source", "state": "file", "uid": 1000}

TASK [6b) Create PVC in k8s cluster] ***********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:85
included: /home/giri/openebs/e2e/ansible/playbooks/utils/deploy_task.yml for localhost

TASK [Deploy application files] ****************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/utils/deploy_task.yml:2
changed: [localhost -> None] => (item=/home/ubuntu/create_pvc.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f /home/ubuntu/create_pvc.yaml -n pvc-deletion", "delta": "0:00:01.082919", "end": "2018-06-12 11:44:11.471476", "item": "/home/ubuntu/create_pvc.yaml", "rc": 0, "start": "2018-06-12 11:44:10.388557", "stderr": "", "stderr_lines": [], "stdout": "persistentvolumeclaim \"openebsvolume\" created", "stdout_lines": ["persistentvolumeclaim \"openebsvolume\" created"]}

TASK [Wait for few seconds] ********************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:91
ok: [localhost -> None] => {"changed": false, "elapsed": 30, "path": null, "port": null, "search_regex": null, "state": "started"}

TASK [7) Finding PV name] **********************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:96
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get pv | grep \"openebs-test\" | awk {'print $1'}", "delta": "0:00:01.005389", "end": "2018-06-12 11:44:44.668987", "rc": 0, "start": "2018-06-12 11:44:43.663598", "stderr": "", "stderr_lines": [], "stdout": "pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db", "stdout_lines": ["pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db"]}

TASK [7a) Confirm whether the volume pods are created] *****************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:104
changed: [localhost -> None] => (item=ctrl) => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n pvc-deletion | grep \"pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db\" | grep ctrl | grep \"Running\" | wc -l", "delta": "0:00:00.611468", "end": "2018-06-12 11:44:45.896490", "item": "ctrl", "rc": 0, "start": "2018-06-12 11:44:45.285022", "stderr": "", "stderr_lines": [], "stdout": "1", "stdout_lines": ["1"]}
changed: [localhost -> None] => (item=rep) => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n pvc-deletion | grep \"pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db\" | grep rep | grep \"Running\" | wc -l", "delta": "0:00:00.612740", "end": "2018-06-12 11:44:46.955174", "item": "rep", "rc": 0, "start": "2018-06-12 11:44:46.342434", "stderr": "", "stderr_lines": [], "stdout": "3", "stdout_lines": ["3"]}

TASK [8) Check if one replica is unscheduled] **************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:119
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n pvc-deletion| grep \"pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db\" | grep -i \"pending\" |wc -l", "delta": "0:00:00.990668", "end": "2018-06-12 11:44:49.188485", "rc": 0, "start": "2018-06-12 11:44:48.197817", "stderr": "", "stderr_lines": [], "stdout": "1", "stdout_lines": ["1"]}

TASK [8a) Check if the svc is created] *********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:131
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get svc -n pvc-deletion | grep \"pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db\" |wc -l", "delta": "0:00:01.257352", "end": "2018-06-12 11:44:51.290560", "rc": 0, "start": "2018-06-12 11:44:50.033208", "stderr": "", "stderr_lines": [], "stdout": "1", "stdout_lines": ["1"]}

TASK [8b) Getting service name] ****************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:142
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get svc -n pvc-deletion| grep \"pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db\" | awk {'print $1'}", "delta": "0:00:00.722438", "end": "2018-06-12 11:44:52.897448", "rc": 0, "start": "2018-06-12 11:44:52.175010", "stderr": "", "stderr_lines": [], "stdout": "pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db-ctrl-svc", "stdout_lines": ["pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db-ctrl-svc"]}

TASK [9) Delete PVC] ***************************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:149
included: /home/giri/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml for localhost

TASK [Delete application] **********************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml:2
changed: [localhost -> None] => (item=create_pvc.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl delete -f create_pvc.yaml -n pvc-deletion", "delta": "0:00:00.613706", "end": "2018-06-12 11:44:54.797475", "item": "create_pvc.yaml", "rc": 0, "start": "2018-06-12 11:44:54.183769", "stderr": "", "stderr_lines": [], "stdout": "persistentvolumeclaim \"openebsvolume\" deleted", "stdout_lines": ["persistentvolumeclaim \"openebsvolume\" deleted"]}

TASK [10) check if the pods are deleted] *******************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:155
FAILED - RETRYING: 10) check if the pods are deleted (10 retries left).
changed: [localhost -> None] => (item=ctrl) => {"attempts": 2, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n pvc-deletion | grep \"pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db\" | grep ctrl | wc -l", "delta": "0:00:00.726446", "end": "2018-06-12 11:47:00.664476", "item": "ctrl", "rc": 0, "start": "2018-06-12 11:46:59.938030", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "0", "stdout_lines": ["0"]}
changed: [localhost -> None] => (item=rep) => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n pvc-deletion | grep \"pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db\" | grep rep | wc -l", "delta": "0:00:00.723492", "end": "2018-06-12 11:47:02.222024", "item": "rep", "rc": 0, "start": "2018-06-12 11:47:01.498532", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "0", "stdout_lines": ["0"]}

TASK [10a) Check if the pv is deleted] *********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:168
 [WARNING]: when statements should not include jinja2 templating delimiters such as {{ }} or {% %}. Found: '{{ pv_name.stdout }}' not in pv_out.stdout

changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pv", "delta": "0:00:00.521927", "end": "2018-06-12 11:47:03.791034", "rc": 0, "start": "2018-06-12 11:47:03.269107", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "", "stdout_lines": []}

TASK [10b) Check if the svc is deleted] ********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:178
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get svc -n pvc-deletion | grep \"pvc-ec76a5cd-6e35-11e8-9ea9-02b983f0a4db-ctrl-svc\" | wc -l", "delta": "0:00:00.292501", "end": "2018-06-12 11:47:04.985270", "rc": 0, "start": "2018-06-12 11:47:04.692769", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "0", "stdout_lines": ["0"]}

TASK [Setting pass flag] ***********************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:188
ok: [localhost] => {"ansible_facts": {"flag": "TEST PASSED"}, "changed": false}

TASK [Delete the namespace] ********************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/cleanup.yml:2
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl delete ns pvc-deletion", "delta": "0:00:00.377091", "end": "2018-06-12 11:47:06.015749", "rc": 0, "start": "2018-06-12 11:47:05.638658", "stderr": "", "stderr_lines": [], "stdout": "namespace \"pvc-deletion\" deleted", "stdout_lines": ["namespace \"pvc-deletion\" deleted"]}

TASK [Delete the storage class] ****************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/cleanup.yml:12
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl delete sc openebs-test", "delta": "0:00:00.348573", "end": "2018-06-12 11:47:07.486750", "rc": 0, "start": "2018-06-12 11:47:07.138177", "stderr": "", "stderr_lines": [], "stdout": "storageclass.storage.k8s.io \"openebs-test\" deleted", "stdout_lines": ["storageclass.storage.k8s.io \"openebs-test\" deleted"]}

TASK [Remove the test artifacts] ***************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/cleanup.yml:18
changed: [localhost -> None] => (item=storage-class.yaml) => {"changed": true, "item": "storage-class.yaml", "path": "/home/ubuntu/storage-class.yaml", "state": "absent"}
changed: [localhost -> None] => (item=create_pvc.yaml) => {"changed": true, "item": "create_pvc.yaml", "path": "/home/ubuntu/create_pvc.yaml", "state": "absent"}

TASK [Setting cleanup flag to passed] **********************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:204
ok: [localhost] => {"ansible_facts": {"cflag": "CLEANUP PASSED"}, "changed": false}

TASK [include_tasks] ***************************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:216
included: /home/giri/openebs/e2e/ansible/playbooks/utils/stern_task.yml for localhost

TASK [Start the log aggregator to capture test pod logs] ***************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/utils/stern_task.yml:2
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [Terminate the log aggregator] ************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/utils/stern_task.yml:14
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; killall stern", "delta": "0:00:00.051596", "end": "2018-06-12 11:47:10.941016", "rc": 0, "start": "2018-06-12 11:47:10.889420", "stderr": "mesg: ttyname failed: Inappropriate ioctl for device", "stderr_lines": ["mesg: ttyname failed: Inappropriate ioctl for device"], "stdout": "", "stdout_lines": []}

TASK [Send slack notification] *****************************************************************************************************************************************
task path: /home/giri/openebs/e2e/ansible/playbooks/hyperconverged/test-k8s-pvc-deletion/k8s_delete_pvc.yml:220
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional result was False"}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************
localhost                  : ok=34   changed=23   unreachable=0    failed=0
```
